app/models/bishop.rb

```diff
@@ -1,5 +1,20 @@
 class Bishop < Piece
 
+	def no_move?(x,y)
+		(x == x_cord) && (y == y_cord) ? true : false
+	end
+
+	def off_board?(x,y)
+		(x > 8 || y > 8 || x < 1 || y < 1) ? true : false
+	end
```

Comment:
Aleks put this method in the peices model:

```ruby
  def position_exist?(x, y)
    x >= 0 && x <= 7 && y >= 0 && y <= 7
  end
```

so we can use position_exist?(x, y) in the bishop_move statement instead of recreating here :)